src/pages/permissions/panels/EditGroupPanel.tsx

```diff
@@ -11,26 +17,110 @@ import { renameGroup, updateGroup } from "api/auth-groups";
 import { queryKeys } from "util/queryKeys";
 import { testDuplicateGroupName } from "util/permissionGroups";
 import NotificationRow from "components/NotificationRow";
-import { LxdGroup } from "types/permissions";
+import { LxdGroup, LxdIdentity } from "types/permissions";
 import ScrollableContainer from "components/ScrollableContainer";
+import { FormPermission } from "pages/permissions/panels/PermissionSelector";
+import classnames from "classnames";
+import EditIdentitiesForm from "pages/permissions/panels/EditIdentitiesForm";
+import { getIdentityIdsForGroup } from "util/permissionIdentities";
+import EditGroupPermissionsForm from "pages/permissions/panels/EditGroupPermissionsForm";
+import { enrichPermissionId } from "util/permissions";
+import { fetchIdentities, updateIdentities } from "api/auth-identities";
+import LoggedInUserNotification from "pages/permissions/panels/LoggedInUserNotification";
+import { useSettings } from "context/useSettings";
+
+export type FormIdentity = {
+  id: string;
+  isRemoved?: boolean;
+  isAdded?: boolean;
+};
 
 interface Props {
   group: LxdGroup;
+  onClose?: () => void;
 }
 
-const EditGroupPanel: FC<Props> = ({ group }) => {
+const EditGroupPanel: FC<Props> = ({ group, onClose }) => {
   const panelParams = usePanelParams();
   const notify = useNotify();
   const toastNotify = useToastNotification();
   const queryClient = useQueryClient();
   const controllerState = useState<AbortController | null>(null);
+  const [subForm, setSubForm] = useState<string | null>(null);
+  const [confirming, setConfirming] = useState(false);
+  const [identities, setIdentities] = useState<FormIdentity[]>(
+    getIdentityIdsForGroup(group).map((id) => ({ id: id })),
+  );
+  const [permissions, setPermissions] = useState<FormPermission[]>(
+    enrichPermissionId(group.permissions ?? []),
+  );
+  const { data: settings } = useSettings();
+  const loggedInIdentityID = settings?.auth_user_name ?? "";
 
   const groupSchema = Yup.object().shape({
     name: Yup.string()
       .test(...testDuplicateGroupName(controllerState, panelParams.group ?? ""))
       .required("Group name is required"),
   });
 
+  const saveIdentities = async () => {
+    const addedIdentities = identities
+      .filter((id) => id.isAdded)
+      .map((id) => id.id);
+    const removedIdentities = identities
+      .filter((id) => id.isRemoved)
+      .map((id) => id.id);
+
+    const identityPayload: LxdIdentity[] = [];
+
+    const existingIdentities = await fetchIdentities();
+    existingIdentities.map((identity) => {
```

Comment:
again I think the logic here can be alot cleaner if we include api data in `identities` state.